docs/HowToGuides/GettingStarted.md

```diff
@@ -0,0 +1,593 @@
+  and looking at the first error may be more helpful than simply looking
+  at the last error.
+- Check if others have encountered the same issue on the Swift forums or on
+  [Swift JIRA][].
+- Create a new Swift forums thread in the Development category. Include
+  information about your configuration and the errors you are seeing.
+  - You can [create a gist](https://gist.github.com) with the entire build
+    output and link it, while highlighting the most important part of the
+    build log in the post.
+  - Include the output of `utils/update-checkout --dump-hashes`.
+
+[Swift JIRA]: https://bugs.swift.org
+
+## Editing code
+
+### First time Xcode setup
+
+If you used `--xcode` earlier, you will see an Xcode project generated under
+`../build/Xcode-RelWithDebInfoAssert/swift-macosx-x86_64`. When you open the
+project, Xcode might helpfully suggest "Automatically Create Schemes". Most of
+those schemes are not required in day-to-day work, so you can instead manually
+select the following schemes:
+- `swift-frontend`: If you will be working on the compiler.
+- `check-swift-all`: This can be used to run the tests. The test runner does
+  not integrate with Xcode though, so it may be easier to run tests directly
+  on the commandline for more fine-grained control over which exact tests are
+  run.
+<!-- TODO: Insert SourceKit/stdlib specific instructions? -->
+
+### Editing
+
+Make changes to the code as appropriate. Implement a shiny new feature!
+Or fix a nasty bug! Update the documentation as you go! <!-- please 🙏 -->
+The codebase is your oyster!
+
+:construction::construction_worker::building_construction:
+
+Now that you have made some changes, you will need to rebuild...
+
+### Incremental rebuilds with Ninja
+
+To rebuild the compiler:
+```sh
+ninja -C ../build/Ninja-RelWithDebInfoAssert/swift-macosx-x86_64 swift-frontend
+```
+
+To rebuild everything, including the standard library:
+```sh
+ninja -C ../build/Ninja-RelWithDebInfoAssert/swift-macosx-x86_64
+```
+
+### Incremental builds with Xcode
+
+Rebuilding works the same way as with any other Xcode project; you can use
+<kbd>⌘</kbd>+<kbd>B</kbd> or Product → Build.
+
+### Spot checking an incremental build
+
+As a quick test, go to `lib/Basic/Version.cpp` and tweak the version
+printing code slightly. Next, do an incremental build as above. This incremental
+build should be much faster than the from-scratch build at the beginning.
+Now check if the version string has been updated:
+
+```sh
+../build/Ninja-RelWithDebInfoAssert/swift-macosx-x86_64/bin/swift-frontend --version
+```
+
+This should print your updated version string.
+
+## Reproducing an issue
+
+Starter bugs typically have small code examples that fit within a single file.
+You can reproduce such an issue in various ways, such as compiling it from the
+commandline using `/path/to/swiftc MyFile.swift`, pasting the code into
+[Compiler Explorer][] (aka godbolt) or using an Xcode Playground.
```

Comment:
We should talk a little bit about `xcrun` and maybe even the `-target` flag here.

https://forums.swift.org/t/how-do-i-debug-the-compiler-when-the-file-needs-to-import-a-framework-like-swiftui/40050